SU2_CFD/src/solvers/CPassiveScalarSolver.cpp

```diff
@@ -0,0 +1,583 @@
+  //su2double   temp_inlet    = config->GetInlet_Ttotal       (Marker_Tag);
+  inlet_scalar  = config->GetInlet_ScalarVal    (Marker_Tag);
+
+  /*--- Loop over all the vertices on this boundary marker ---*/
+
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+
+    auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+
+    if (geometry->nodes->GetDomain(iPoint)) {
+
+      nodes->SetSolution_Old(iPoint, inlet_scalar);
+
+      LinSysRes.SetBlock_Zero(iPoint);
+
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        nodes->SetVal_ResTruncError_Zero(iPoint, iVar);
+      }
+
+      /*--- Includes 1 in the diagonal ---*/
+      for (auto iVar = 0u; iVar < nVar; iVar++) {
+        auto total_index = iPoint*nVar+iVar;
+        Jacobian.DeleteValsRowi(total_index);
+      }      
+    }
+  }
+
+}
+
+void CPassiveScalarSolver::BC_Outlet(CGeometry *geometry,
+                                     CSolver **solver_container,
+                                     CNumerics *conv_numerics,
+                                     CNumerics *visc_numerics, CConfig *config,
+                                     unsigned short val_marker) {
+ 
+  /*--- Loop over all the vertices on this boundary marker ---*/
+  
+  for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    
+    /* strong zero flux Neumann boundary condition at the outlet */
+    auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+    
+    /*--- Check if the node belongs to the domain (i.e., not a halo node) ---*/
+    
+    if (geometry->nodes->GetDomain(iPoint)) {
+      
+        /*--- Allocate the value at the outlet ---*/
+        auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor(); 
+          
+        for (auto iVar = 0u; iVar < nVar; iVar++)
+          Solution[iVar] = nodes->GetSolution(Point_Normal, iVar);
+        nodes->SetSolution_Old(iPoint, Solution);
+    
+        LinSysRes.SetBlock_Zero(iPoint);
+
+        for (auto iVar = 0u; iVar < nVar; iVar++){
+          nodes->SetVal_ResTruncError_Zero(iPoint, iVar);
+        }
+
+        /*--- Includes 1 in the diagonal ---*/
+        for (auto iVar = 0u; iVar < nVar; iVar++) {
+          auto total_index = iPoint*nVar+iVar;
+          Jacobian.DeleteValsRowi(total_index);
+        }
+    }
+  }
+  
+}
+
+void CPassiveScalarSolver::BC_Isothermal_Wall(CGeometry *geometry,
+                                              CSolver **solver_container,
+                                              CNumerics *conv_numerics,
+                                              CNumerics *visc_numerics,
+                                              CConfig *config,
+                                              unsigned short val_marker) {
+  
+  /*--- Convective fluxes across viscous walls are equal to zero. ---*/
+  
+}
```

Comment:
What is the reason to handle only inlet and outlet BC's in the passive and scalar solvers?

Reply:
For walls, we assume zero flux for the passive (nonreacting) scalars, so we don't have to do anything (I hope?). For combustion, we can impose a temperature, which is then converted to an enthalpy boundary condition for the enthalpy equation (one of the transported scalars is the equation for total enthalpy).
Other boundaries we haven't considered yet.

Reply:
Then you are assuming all walls are "adiabatic" otherwise you have a direct contribution to the residual from a boundary flux, or one computed from the normal gradient in the isothermal (Dirichlet) case.